sdk/core/core-rest-pipeline/review/core-rest-pipeline.api.md

```diff
@@ -38,6 +55,25 @@ export interface BearerTokenAuthenticationPolicyOptions {
     scopes: string | string[];
 }
 
+// @public
+export function bearerTokenChallengeAuthenticationPolicy(options: BearerTokenChallengeAuthenticationPolicyOptions): PipelinePolicy;
+
+// @public
+export const bearerTokenChallengeAuthenticationPolicyName = "bearerTokenChallengeAuthenticationPolicy";
+
+// @public
+export interface BearerTokenChallengeAuthenticationPolicyOptions {
+    challengeCallbacks?: ChallengeCallbacks;
+    credential: TokenCredential;
+    scopes: string[];
+}
+
+// @public
+export interface ChallengeCallbacks {
+    authorizeRequest?(options: AuthorizeRequestOptions): Promise<void>;
```

Comment:
I wonder if we should rename this to `authorizeRequestBeforeSend` or something to make it clear when it gets called in the naming? Otherwise, it's difficult to know its function without reading dosc (which not all devs will do)

Reply:
This name is picked for x-language consistency. I think it's an advanced scenario to implement CAE so I'd expect developers to learn more about it.